libpkgconf/pkg.c

```diff
@@ -119,9 +119,28 @@ pkgconf_pkg_dir_list_build(pkgconf_client_t *client)
 
 	if (!(client->flags & PKGCONF_PKG_PKGF_ENV_ONLY))
 	{
+#ifdef __HAIKU__
```

Comment:
i'm not sure this is the way to go here.  should we maybe introduce more abstraction?

Reply:
Yes, that would be fine.